### PR TITLE
Add additional tests to App

### PR DESCRIPTION
**Pull request by foxwellm**

### CODE REVIEW

File: src/components/NotFound/NotFound.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React from 'react'
 import { Link } from 'react-router-dom'
 
-const NotFound = () => {
+export const NotFound = () => {
```
Comment (geet084) [inline src/components/NotFound/NotFound.js:4]: I always forget to do that until the test, glad you fixed that

File: src/containers/App/App.js [hunk 2]
```diff
@@ -34,10 +33,12 @@ class App extends Component {
               }} />
               <Route path='/notes/:id' render={({ match }) => {
                 const { id } = match.params
-                const note = this.props.notes.find(note => note.id === id)
+                const note = this.props.notes.find(note => note.id == id)
```
Comment (geet084) [inline src/containers/App/App.js:36]: why did you have to change this? I thought it was working with the strictly equals?

File: src/containers/App/App.test.js [hunk 3]
```diff
@@ -39,16 +41,69 @@ describe('App', () => {
...
+
+    it('should render the NoteArea container when at the root route', () => {
+      const wrapper = mount(
+        <Provider store={store}>
+          <MemoryRouter initialEntries={['/']}>
+            <App />
+          </MemoryRouter>
+        </Provider>
+      )
+      expect(wrapper.find(NoteArea)).toHaveLength(1)
```
Comment (geet084) [inline src/containers/App/App.test.js:59]: that's slick, I never would have thought of that

File: src/containers/App/App.test.js [hunk 4]
```diff
@@ -39,16 +41,69 @@ describe('App', () => {
...
+      const fetchNotes = jest.fn()
+      wrapper = mount(
+        <Provider store={store}>
+          <MemoryRouter initialEntries={['/notes/1']}>
+            <App notes={mockNotes} items={mockItems} error={''} fetchNotes={fetchNotes} />
+          </MemoryRouter>
+        </Provider>
+      )
+      expect(wrapper.find(NoteForm)).toHaveLength(1)
+    })
```
Comment (geet084) [inline src/containers/App/App.test.js]: I don't quite understand, what does this prove? in the next test you have the same expectation...?

**Review by geet084 — APPROVED** [review]

File: src/containers/App/App.test.js (hunk 3 above)
Comment (ericweissman) [reply src/containers/App/App.test.js:59]: Yeah I have also not seen it tested this way - will definitely be sure to reference this on our next project!

**Review by ericweissman — APPROVED** [review]

File: src/containers/App/App.js (hunk 2 above)
Comment (foxwellm) [reply src/containers/App/App.js:36]: for some reason the test file is seeing the id as a string when using the mock data

**Pull request merged**